### PR TITLE
Add support for job URLs as job spec

### PR DESCRIPTION
**Pull request by eliasdorneles**
With this PR, you can do:

    shub log -f <PASTE DASH JOB LINK HERE>

Beats typing or editing the link to supply only the job key every time. ;)

### CODE REVIEW

**Comment by pablohoffman** [comment]
+1 functionality-ise

File: tests/test_utils.py [hunk 1]
```diff
@@ -113,6 +113,8 @@ def _test_specs(job, expected_job_id, expected_endpoint):
         _test_specs('default/2/3', '1/2/3', 'default')
         _test_specs('prod/2/3', '2/2/3', 'default')
         _test_specs('vagrant/2/3', '3/2/3', 'vagrant')
+        _test_specs('https://dash.scrapinghub.com/p/7389/job/259/1/#/log/line/0',
```
Comment (jdemaeyer) [inline tests/test_utils.py]: Could you shorten this line to 79 characters? I'm a little pedantic about PEP8, sorry :D

Comment (eliasdorneles) [inline tests/test_utils.py]: haha, will do :)

**Comment by jdemaeyer** [comment]
Very nice feature, thanks!

Could you mention it in the `HELP` texts in `log.py`/`items.py`/`requests.py`? (There's currently a lot of repetition in these, will be fixed)

**Comment by eliasdorneles** [comment]
Sure thing, gimme a min

**Pull request merged**